src/main/java/htsjdk/samtools/cram/encoding/reader/MultiRefSliceAlignmentSpanReader.java

```diff
@@ -64,24 +62,41 @@ public MultiRefSliceAlignmentSpanReader(final BitInputStream coreInputStream,
         this.prevAlignmentStart = initialAlignmentStart;
 
         for (int i = 0; i < recordCount; i++) {
-            readCramRecord();
+            final CramCompressionRecord cramRecord = new CramCompressionRecord();
+            prevAlignmentStart = super.read(cramRecord, prevAlignmentStart);
+            processRecordSpan(cramRecord);
         }
     }
 
     public Map<ReferenceContext, AlignmentSpan> getReferenceSpans() {
         return Collections.unmodifiableMap(spans);
     }
 
-    private void readCramRecord() {
-        final CramCompressionRecord cramRecord = new CramCompressionRecord();
+    private void processRecordSpan(final CramCompressionRecord cramRecord) {
+        // if unplaced: create or replace the current spans map entry.
+        // we don't need to combine entries for different records because
+        // we count them elsewhere and span is irrelevant
 
-        prevAlignmentStart = super.read(cramRecord, prevAlignmentStart);
+        if (! cramRecord.isPlaced()) {
+            spans.put(ReferenceContext.UNMAPPED_UNPLACED_CONTEXT, AlignmentSpan.UNPLACED_SPAN);
```

Comment:
So, do we not count (keep track of) the number of records in UNMAPPED_UNPLACED_CONTEXT ?

Reply:
The Slice itself holds the mapped, unmapped, and unplaced counts.  We need to distinguish between the different references for mapped and unmapped counts - and that is the true (sole??) purpose of both AlignmentSpan and MultiRefSliceAlignmentSpanReader.

We can get unplaced counts by querying the Slice directly, so AlignmentSpan can ignore this.